### ISSUE TITLE
Will Zappa still be working with AWS State Changes?

### ISSUE THREAD

**Issue by lightmagic1** [zappa/Zappa#1058]
"
Hello,

You are receiving this message because we identified that your account created or updated or invoked Lambda functions on or after July 1, 2021.

AWS Lambda is extending the capability to track the current state of a function through its lifecycle to all functions [1]. With this change, you may need to update your CLI and/or SDK-based automation workflows around creating and updating functions by adding a check that the function became active before performing additional actions that operate on the function.

Previously, states have been used in two primary use-cases. Firstly, to move the blocking setup of VPC resources out of the path of function invocation. Secondly, to allow the Lambda service to optimize new or updated container images for container-image based functions, also before invocation. By moving this additional work out of the path of the invocation, customers see lower latency and better consistency in their function performance.
Infrastructure as code and deployment automation tools such as AWS CloudFormation, AWS Serverless Application Model (SAM), AWS Cloud Development Kit (CDK), Serverless Framework,Hashicorp Terraform, AWS Chalice and Cloud Custodian already support states.

If you are using these tools and are already on the latest Software Development Kit (SDK), you do not need to take any action, and can stop reading now. If you are using an earlier SDK version, please update to the latest one.

We are rolling out this change over multi-phase periods to allow you to update existing tooling for deploying and managing Lambda functions. You can delay this change for your functions until December 5, 2021. Starting December 6, 2021, the delay mechanism expires and all customers see the Lambda states lifecycle applied during function create or update. Read this blog post [2] to learn about this change, timelines for different phases and a reference example on how to check your function state.

If you have any questions, please contact AWS Support [3].

[1] https://docs.aws.amazon.com/lambda/latest/dg/functions-states.html
[2] https://aws.amazon.com/blogs/compute/coming-soon-expansion-of-aws-lambda-states-to-all-functions
[3] https://aws.amazon.com/support

Sincerely,
Amazon Web Services
"

Are we going to need to do something or update zappa before Dec 5? Any impacts that i should worry about?

**Comment by paulclarkaranz** [comment]
I believe this is resolved by https://github.com/zappa/Zappa/pull/992 (included in that latest Zappa release, 0.54.0).
I tried opting in to the upcoming change by editing a lambda description as described in the blog post and with **0.50.0** I got:

```
botocore.errorfactory.ResourceConflictException: An error occurred (ResourceConflictException) when calling the UpdateFunctionConfiguration operation: The operation cannot be performed at this time. An update is in progress for resource: arn:aws:lambda:us-west-1:12345678:function:myfunc-dev
```
When running zappa update.

I upgraded to **0.54.0** and it deployed fine.

TL;DR: If you are worried you can temporarily opt in to the change and test for yourself. If my test is anything to go by you should upgrade to the latest version (0.54.0 went out on October 15th) if you haven't already.

**Comment by hellno** [comment]
Thanks, happy that Zappa seems to be compatible with the recent changes.

Can anyone confirm the test by @paulclarkaranz so we can close this issue?

**Comment by javulticat** [comment]
@hellno I can confirm that the issue mentioned is resolved by `v0.54.0` - closing.

**Comment by npbjr** [comment]
Hi Zappa team, how can i make this work in python 2?
im getting this
**RuntimeError: This version of Python (2.7) is not supported!
Zappa (and AWS Lambda) support the following versions of Python: ['3.6', '3.7', '3.8']**
i cant update just yet to python 3

**Comment by javulticat** [comment]
@npbjr Zappa no longer supports Python 2.7, and we do not intend to support it in the future.  The last version of `zappa` to support Python 2 was [`0.48.2`](https://github.com/zappa/Zappa/blob/0.54.1/CHANGELOG.md#0482) - if necessary, you could try creating your own fork of `0.48.2` and adding functionality similar to #992, but we strongly recommend that you upgrade to Python 3 - Python 2 was [sunset](https://www.python.org/doc/sunset-python-2/) by the Python Software Foundation over 2 years ago and no longer receives security patches.

**Comment by npbjr** [comment]
@javulticat thank you for the quick response
